components/Main.vue

```diff
@@ -1,8 +1,274 @@
 <template>
-  <v-row>
-    <v-col class="text-center text-h3">
-      Let's go!
-      <v-icon color="success" icon="mdi-rocket-launch-outline" size="x-small" />
+  <v-row :class="[$vuetify.theme.dark ? 'currency-converter dark-theme' : 'currency-converter light-theme']">
+    <v-col cols="12" class="text-center pb-0">
+      <div class="d-flex">
+        <img src="../docs/logo-cd.png" alt="logo-accueil">
+        <h1>Convertisseur de devises</h1>
+      </div>
+      <div class="converter">
+        <v-text-field v-model="amount" label="Montant" type="number" class="amount" :rules="[v => v >= 0 ]" outlined />
+        <v-select v-model="fromCurrency" :items="currencyOptions" label="De" class="de" outlined />
+        <v-icon @click="swapCurrencies" class="icon-change" color="white">
+          mdi-swap-horizontal
+        </v-icon>
+        <v-select v-model="toCurrency" :items="currencyOptions" label="Vers" class="vers" outlined />
+      </div>
+      <div v-if="result" class="conversion-result mt-3">
+        <h4>{{ formattedResult }}</h4>
+        <p>{{ currentDate }}</p>
+      </div>
+    </v-col>
+    <v-col cols="12" class="switch-theme">
+      <v-switch v-model="$vuetify.theme.dark" label="Sombre/Clair" class="mt-4"></v-switch>
     </v-col>
   </v-row>
+  <div class="table">
+    <h2>Historique des conversions</h2>
+    <v-simple-table>
+      <thead>
+        <tr>
+          <th>Date</th>
+          <th>Montant</th>
+          <th>De</th>
+          <th>Vers</th>
+          <th>Résultat</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="(conversion, index) in conversionHistory" :key="index">
+          <td>{{ conversion.date }}</td>
+          <td>{{ conversion.amount }}</td>
+          <td>{{ conversion.fromCurrency }}</td>
+          <td>{{ conversion.toCurrency }}</td>
+          <td>{{ conversion.result }}</td>
+        </tr>
+      </tbody>
+    </v-simple-table>
+  </div>
 </template>
+
+<script>
+export default {
+  // Object Properties
+  data() {
+    return {
+      amount: 1,
+      fromCurrency: 'EUR',
+      toCurrency: 'USD',
+      exchangeRates: null,
+      result: null,
+      currentDate: this.getCurrentDate(), // Initialize with the current date and time
+      conversionHistory: [], // Table to store conversion history
+      initialized: false, // Flag to skip first conversion
+    };
+  },
+  // Calculated properties of the object
+  computed: {
+    // Returns the currency options available for conversion
+    currencyOptions() {
+      return Object.keys(this.exchangeRates || {}).map(currency => currency.toUpperCase());
```

Comment:
Ça fonctionne.

Dans le cas ou les devises n'avaient pas leur code en minuscule en guise de clé mais un index numérique, il aurait été plus propre d'utiliser les valeurs des objets, au moins on est sûr du code fourni.

```suggestion
      return Object.values(this.exchangeRates || {}).map(currency => currency.code);
```

C'est bien d'avoir pensé au cas ou la liste est vide (je pense que ça ne fonctionnerait pas sans ça).